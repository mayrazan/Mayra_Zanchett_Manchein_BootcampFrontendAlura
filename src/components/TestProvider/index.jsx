import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../../theme';
import GlobalStyle from '../../theme/GlobalStyle';

const TestProvider = ({ children }) => (
  <ThemeProvider theme={theme}>
    <GlobalStyle />
    {children}
  </ThemeProvider>
);

TestProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TestProvider;