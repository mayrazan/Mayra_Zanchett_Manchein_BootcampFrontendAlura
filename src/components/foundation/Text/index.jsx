import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import propToStyle from '../../../theme/utils/propToStyle';
import breakpointsMedia from '../../../theme/utils/breakpointsMedia';

// eslint-disable-next-line consistent-return
export const TextStyleVariants = (name) => {
  if (name)
    return css`
      font-size: ${({ theme }) => theme.typographyFonts[name].fontSize};
      font-weight: ${({ theme }) => theme.typographyFonts[name].fontWeight};
      line-height: ${({ theme }) => theme.typographyFonts[name].lineHeight};
    `;
};

const TextBase = styled.span`
  ${propToStyle('textAlign')}
  ${propToStyle('margin')}
  color: ${({ theme, color }) => color && theme.colors.primary.main[color]};
  ${({ variant }) => {
    if (typeof variant === 'string') {
      return TextStyleVariants(variant);
    }
    return breakpointsMedia({
      xs: css`
        ${TextStyleVariants(variant.xs)}
      `,
      md: css`
        ${TextStyleVariants(variant.md)}
      `,
    });
  }}
  ${({ as }) =>
    as === 'a' &&
    css`
      cursor: pointer;
    `}
`;

export default function Text({ variant, children, tag, color, ...props }) {
  return (
    <TextBase
      as={tag}
      variant={variant}
      color={color}
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...props}
    >
      {children}
    </TextBase>
  );
}

Text.propTypes = {
  tag: PropTypes.string,
  variant: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  children: PropTypes.node.isRequired,
  color: PropTypes.string,
};

Text.defaultProps = {
  tag: 'span',
  variant: 'paragraph1',
  color: '#000000',
};
