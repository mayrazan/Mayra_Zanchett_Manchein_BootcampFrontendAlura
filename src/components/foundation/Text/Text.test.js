import React from 'react';
import { render, screen } from '@testing-library/react';
import Text from '.';
import TestProvider from '../../TestProvider';

describe('<Text />', () => {
  test('renderiza componente com props', () => {
    render(
      <TestProvider>
        <Text
          tag="a"
          variant={{ md: 'menuText' }}
          color="contrastText"
          href="/aboutme"
          bold
          textAlign="center"
        >
          Sobre Mim
        </Text>
      </TestProvider>,
    );
    const text = screen.getByRole('link', {
      name: /sobre mim/i,
    });
    expect(text).toMatchSnapshot();
  });

  test('renderiza componente', () => {
    render(
      <TestProvider>
        <Text>Sobre Mim</Text>
      </TestProvider>,
    );

    const text = screen.getByText(/sobre mim/i);
    expect(text).toMatchSnapshot();
  });
});
