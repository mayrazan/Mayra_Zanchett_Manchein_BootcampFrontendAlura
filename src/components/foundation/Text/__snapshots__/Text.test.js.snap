// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renderiza componente 1`] = `
.c0 {
  font-size: 18px;
  font-weight: normal;
  line-height: 22px;
}

<span
  class="c0"
  color="#4e4e4e"
>
  Sobre Mim
</span>
`;

exports[`<Text /> renderiza componente com props 1`] = `
.c0 {
  color: inherit;
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
  opacity: 1;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
}

.c0:hover,
.c0:focus {
  opacity: 0.5;
}

.c1 {
  text-align: center;
  color: #ffffff;
}

@media screen and (min-width:0px) {

}

@media screen and (min-width:768px) {
  .c1 {
    font-size: 28px;
    font-weight: bold;
    line-height: 34px;
  }
}

<a
  class="c0 c1"
  color="contrastText"
  href="/aboutme"
  rel="noreferrer"
>
  Sobre Mim
</a>
`;
